src/js/components/Button/Button.js

```diff
@@ -86,14 +86,27 @@ class Button extends Component {
     } = this.props;
     const { hover } = this.state;
 
-    let buttonIcon = icon;
-    // only change color if user did not specify the color themselves...
-    if (primary && icon && !icon.props.color) {
+    let buttonIcon;
+    let buttonColor;
```

Comment:
Can we move this inside the if statement at line 91? That would make it clearer that it doesn't have side effects outside of the if statement. Also, I wonder if we should rename the variable, `buttonColor` sounds like it will be the color of the button, but I think we only using it to set the icon color.